lib/awesome_print/formatter.rb

```diff
@@ -318,6 +318,26 @@ def plain_single_line
 
     # Utility methods.
     #------------------------------------------------------------------------------
+    def convert_to_hash(object)
+      if not object.respond_to?(:to_hash)
+        return nil
+      end
+      if object.method(:to_hash).arity != 0
+        return nil
+      end
+
+      hash = object.to_hash
+      if not hash
+        return nil
+      end
+
+      if not hash.respond_to?(:keys) or not hash.respond_to?('[]')
```

Comment:
Prefer the `||` over the `or`, take a look at https://github.com/bbatsov/ruby-style-guide#no-and-or-or